lib/fetchContent.js

```diff
@@ -323,6 +324,141 @@ function getRecentArticles(days, aspects, facets)
 		});
 }
 
+function getArticleTopics(days, aspects, facets)
+{
+	const date = time.getDatetimeRange('days', days, 0);
+	const params = {
+		queryString: `lastPublishDateTime:> ${date.next}`,
+		maxResults : 100,
+		aspects : aspects,
+		facets : {
+			"names": facets,
+			"maxElements":-1
+		}
+	};
+
+	return searchDeeper(params, 10)
+		.then(results =>  extractArticles(results) )
+		.then(articles => sortArticlesIntoTopics(articles) )
+		.catch(err => {
+			throw err;
+		});
+}
+
+function sortArticlesIntoTopics(articles)
+{
+	let allTopics = [];
+	let uniqueTopics = [];
+	let topicsWithArticles = [];
+
+	// Get all topics covered by articles
+	articles.forEach(article => {
+		var topics = article.metadata.topics;
+
+		if(topics && topics !== undefined){
+			topics.forEach(topic => {
+				allTopics.push(topic.term.name);
+			});
+		}
+	});
+
+	uniqueTopics = array.unique(allTopics);
+
+	//console.log(uniqueTopics);
+
+
+	// Add all topics to final list
+	uniqueTopics.forEach(topic => {
+		topicsWithArticles.push({
+			name: topic,
+			articles: [],
+			articleCount: 0,
+			relatedTopics: [],
+		});
+	});
+
+	//console.log(topicsWithArticles);
+
+
+	// Add articles to their matching topics (in final list)
+	articles.forEach(article => {
+		var topics = article.metadata.topics;
+
+		if(topics && topics !== undefined){
+			topics.forEach(topic => {
+
+				let name = topic.term.name;
+				let tempList = [];
+				
+				tempList.push(article);
+
+				var obj = topicsWithArticles.find(function (obj) { return obj.name === name; });
+
+				if(obj && obj.articles.length >= 1){
+
+					let currentArticles = obj.articles;
+					currentArticles.push(article);
+					tempList = currentArticles;
+
+					obj.articleCount = obj.articleCount + 1;
+					obj.articles = tempList;
+
+
+					let topicHolder = [];
+					let moarTopics = article.metadata.topics;
+
+					moarTopics.forEach(topic => {
+						topicHolder.push(topic.term.name);
+					});
+
+
+					let mergedTopics = topicHolder.concat(obj.relatedTopics);
+					let uniqueTopics = array.unique(mergedTopics);
+
+					obj.relatedTopics = uniqueTopics;
+
+
+
+				} else {
+					obj.articleCount = 1;
+					obj.articles = tempList;
+
+
+					let topicHolder = [];
+					let moarTopics = article.metadata.topics;
+
+					moarTopics.forEach(topic => {
+						topicHolder.push(topic.term.name);
+					});
+
+					obj.relatedTopics = topicHolder;
+				}
+
+			});
+		}
+	});
+
+
+	//console.log(topicsWithArticles);
+
+	
+
+
+	articles = topicsWithArticles;
+
+	
+
+
```

Comment:
Could do with reducing some of the whitespace

Reply:
have removed